src/CMakeLists.txt

```diff
@@ -255,8 +255,15 @@ if(CCTAG_VISUAL_DEBUG)
 endif(CCTAG_VISUAL_DEBUG)
 if(CCTAG_EIGEN_NO_ALIGN)
   target_compile_definitions(CCTag PUBLIC ${AV_EIGEN_DEFINITIONS})
+else()
+  # If user enabled Eigen alignment assumptions, then allocations should be with appropriate
+  # alignment. Fortunately this is fixed in C++17. While we can't upgrade to C++17 just yet, some
+  # compilers support overaligned allocation feature with a separate flag.
+  # See https://eigen.tuxfamily.org/dox/group__TopicUnalignedArrayAssert.html
+  if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.1)
```

Comment:
shall we also check eigen version? it seems like it is supported after 3.4?
https://gitlab.com/libeigen/eigen/-/issues/1409

Reply:
It's enough just to specify `-faligned-new` to force the new operator to take into account alignment. I've just checked this assumption with this example:

```

#include <iostream>

struct X {
    __attribute__((aligned(256))) int a[64];
};

int main()
{
    X x;
    X* y = new X;
    std::cout << std::hex << reinterpret_cast<std::uintptr_t>(&x) << "\n";
    std::cout << std::hex << reinterpret_cast<std::uintptr_t>(y) << "\n";
    delete y;
}
```

`__attribute__((aligned(x)))` is what Eigen uses to force alignment.

 When building with `g++ -std=c++14 aligned_new.cc` and running I get the following:

```
7fffffffda00
55555556aeb0
```

Data is correctly aligned on stack, which is expected because the compiler has full control of the alignment on stack and has enough information. On the other hand, allocation from heap requires library support and since there's no aligned new in C++14 mode we get unaligned pointer. 

When building with `g++ -std=c++14 -faligned-new aligned_new.cc` or `g++ -std=c++17 aligned_new.cc` and running I get the following in both cases:

```
7fffffffda00
55555556af00
```

Both stack and heap allocations are aligned without any additional work by the developer.